### ISSUE TITLE
Option to suppress bounds exception on incompressibles

### ISSUE THREAD

**Issue by paarfi** [CoolProp/CoolProp#576]
Is there a way (or could you add one) to suppress the temperature out of bounds error on incompressible calculations?
Your temperature 273.000000 is not between 273.150000 and 473.150000.

For evaporator calculations, your final answer may be above the freeze point (or within the usable bounds) of the secondary fluid, but the iterations needed to arrive at that final answer may drift lower. If an exception is raised there, then the whole iteration fails before it arrives at a good solution. The calculated values out of bounds may not be accurate, but as long as the curves are reasonably continuous the iteration should continue to converge. For those situations, it would be nice if there were a flag on the AbstractState to tell it to ignore the bounds checks and let the caller handle that check at the end.

**Comment by ibell** [comment]
I'm definitely sympathetic to your plight.  I'm not sure how easy it would be though, @jowr would have to chime in here.  If we use a bounded solver, it's not possible, if its a secant-based solver, might be possible.

**Comment by jowr** [comment]
For inputs other than T,p we use a bounded solver. Nothing can be done. However, it looks like you use T as an input. In that case, we might be able to suppress the warning/exception. Any ideas on how to implement this without messing up the fluid structure?

**Comment by paarfi** [comment]
Yes, this would be with T as the input. I'm not familiar enough with your internal structure to comment on the best way to implement it though. Hopefully you can find something that works well. Thanks.

**Comment by ibell** [comment]
You could also just extrapolate yourself if you are out of range.  That's a better solution probably for your case than adjusting the bounds.  You could get two points near the bottom limit and then just linearly extrapolate.  Would work in the evaporator model, and hopefully your solution is back within range.

**Comment by paarfi** [comment]
That's a good idea. Thanks.

**Comment by ibell** [comment]
Here are a few ideas:

a) Extrapolate, as I mention above.  Only enabled if ``IGNORE_LIMITS`` configuration variable is set
b) Use a newton-based solver, covering the min,max range.  Does not ensure solution is within range, but might allow for some extrapolation.  Only enabled if ``IGNORE_LIMITS`` configuration variable is set
c) Implement b) and allow user to provide guess value using ``update_with_guesses``.  They are then on their own and responsible for whatever comes out.

**Comment by paarfi** [comment]
I've implemented the extrapolation idea like you had suggested before. It seems to do the trick and has solved my immediate problem. I suppose other people with the same issue could do the same thing. So I guess it's a question of whether having that as part of coolprop itself is better or more elegant, and whether it is worth the added interface complexity and code work. I guess I could go either way on that.

**Comment by Jeffersonjjsk** [comment]
@paarfi I have followed this thread with interest. I have modified ACHP to model a chiller using a plate HEX for the evaporator. All works fine until the evaporator temperature falls below 0 degC, at which point I get the error "ValueError: Your temperature 268.551551 is not between 273.150000 and 473.150000." This is a limitation when modelling chillers that I need to solve, as water temperatures can be as low as 4C. I  am interested in adopting the method (extrapolation) you used although I am not too familiar with C++(I'm ok with python) to make the change in Coolprop or ACHP(I'm not sure which). I can get a friend to make the change to C++ if you could send me the required changes. Thanks in anticipation.

**Comment by paarfi** [comment]
@Jeffersonjjsk Well, all I did was put a wrapper in my code around the calls to CoolProp. In my case, I always know T and I'm trying to calculate the other parameters. 

First I find the minimum temperature (Tmin) that the CoolProp calcs will support. These are listed here for the pure fluids: http://www.coolprop.org/fluid_properties/Incompressibles.html
and you can use CoolProp to calculate T_freeze for the brines based on your concentration.

Then I can compare the T I am interested in to the Tmin. If (T > Tmin) I can call CoolProp and trust that it will return me a good answer. If (T <= Tmin) then we have to do something more complicated.

To get "reasonable" values in that 2nd case, I use CoolProp to calculate the property I am interested in at two other temperatures: (Tmin + 0.01) and (Tmin + 1). Then I use linear interpolation to find the property I'm looking for at my T based on those two other points. So if I'm looking for Cp, for example, it would look like this:

`if (T > Tmin)`
`  Cp = FindCpUsingCoolProp(T)`
`else`
`  T0 = Tmin + 0.01`
`  T1 = Tmin + 1`
`  Cp0 = FindCpUsingCoolProp(T0)`
`  Cp1 = FindCpUsingCoolProp(T1)`
`  Cp = Cp0 + (Cp1 - Cp0) * (T - T0) / (T1 - T0)`
`endif`

It is not terribly elegant and not something they would want to put into CoolProp itself, but It did address my issues with iterations wandering below Tmin before finishing above. It is important to check that your T is actually higher than Tmin once your iterations are done though, especially if Tmin is the freeze point of the fluid.

Hope that helps.

**Comment by ibell** [comment]
Closing as part of an issue-tracker cleanup — this issue has had no activity in over two years.

*To be transparent: this comment and the close action are being performed by Claude (Anthropic's AI coding assistant), running on behalf of @ibell. Ian reviewed and approved the bulk-close criteria; I am the one executing the action.*

If you would like this issue reconsidered, please feel free to **re-open it** — a fresh comment with current behaviour against the latest CoolProp release is the most useful way to revive the discussion.

— Claude (acting for @ibell)
